### ISSUE TITLE
Neovim and Helix rendering with line spacing

### ISSUE THREAD

**Issue by viirak** [raphamorim/rio#398]
<img width="346" alt="image" src="https://github.com/raphamorim/rio/assets/14965/ed3f17ff-375b-4915-bfea-b762aa22b633">
As seen in the screenshot. There's lines between each line. It is clearly visible with light theme. Note sure what's wrong here? 🤔

Any idea?

**Comment by TornaxO7** [comment]
May I ask for your `helix` config?

**Comment by SteveMcGrath** [comment]
The acme theme seems to do it: 
![CleanShot 2024-01-31 at 10 05 45](https://github.com/raphamorim/rio/assets/254700/fa2753a4-bec7-48b1-8b48-85b28892f0bd)

**Comment by raphamorim** [comment]
@viirak @SteveMcGrath thanks for the input, question are you folks using the default font family?

**Comment by SteveMcGrath** [comment]
Here is using the default font: 
![CleanShot 2024-01-31 at 10 48 17](https://github.com/raphamorim/rio/assets/254700/442fe96d-d446-4044-b933-d83fa0c41947)

I've seen other oddities as well with some ligatures being represented weirdly as well.  wonder if its related.

rio (on top) vs iterm2 (below)

![CleanShot 2024-01-31 at 10 50 42](https://github.com/raphamorim/rio/assets/254700/b49eb550-3366-4e18-9da3-f4f6f3a05c39)

**Comment by SeleneCosmia** [comment]
This happened for me as well until I turned off window background transparency. Setting it to be completely opaque removes this rendering issue.

**Comment by GreatGodOfFire** [comment]
The problem is that rio is drawing overlapping rectangles for the background of characters. Due to the transparency the overlapping regions are less transparent, making them visible, unless the background opacity is set to 0 or 1. But it is also visible if its completely opaque in special cases, like in this sugarloaf example
![image](https://github.com/raphamorim/rio/assets/53192467/e0fd8225-efb8-4f7a-9626-743c90c1866c)
where the underline is partially blocked by the background of the second line.

**Comment by raphamorim** [comment]
This issue should not be able to reproduce with the latest release (v0.1.0), closing this issue for now. In case find anything odd we can reopen or create a new issue.
